feat(max): floating Max drag&snap v2

kappa90 opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Problem
The current "drag&snap" animation has hardcoded values and doesn't work properly with the resizing panels. This PR fixes that and adds a nice "morphing" animation when switching between floating Max and input.

## Changes
- Refactor "drag&snap" hook & positioning utils
- Better "morphing" animation
- Snaps to the other side if moved above 100px on the X axis

https://github.com/user-attachments/assets/bb50b3f6-fdfc-4672-87ef-0c42f1fe2bd5

## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [X] No docs needed for this change

## How did you test this code?
Added tests + tested visually

Commit:
feat: drag & snap v2

Commit:
fix: v2

Commit:
feat: morphing animation

Commit:
fix: remove useless test

Commit:
feat: drag to other side if above threshold

github-actions[bot] committed:
Update UI snapshots for `chromium` (9)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

Commit:
Merge remote-tracking branch 'origin/master' into feat/max-snap-2.0

Twixes reviewed (changes requested):
Bits we talked about:
- with this animation, the expanded mode visually expands from its center, rather than bottom right corner – the previous animation was smoother that way
- once dragging end, the rounding on the collapsed mode goes AWOL
- it's okay not to have a dragging affordance on the collapsed mode, but we should really make the expanded one draggable

Commit:
Merge branch 'master' into feat/max-snap-2.0

Commit:
fix: drag input as well

Commit:
fix: refactor / simplify components

github-actions[bot] committed:
Update UI snapshots for `chromium` (14)

github-actions[bot] committed:
Update UI snapshots for `chromium` (13)

github-actions[bot] committed:
Update UI snapshots for `chromium` (10)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Twixes committed:
Fix dragging style

Dragging had two issues:
1. Backdrop blur disappeared.
2. It snapped without including the mb-2 bottom offset, resulting in a janky glitch.

Twixes committed:
Don't show "Drag to move" tooltip while dragging

Twixes committed:
Merge branch 'master' into feat/max-snap-2.0

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (16)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Twixes reviewed (approved):
🛟

github-actions[bot] committed:
Update UI snapshots for `chromium` (16)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)
